libcaf_net/caf/net/web_socket/framing.cpp

```diff
@@ -10,6 +10,49 @@
 
 namespace caf::net::web_socket {
 
+// -- static utility functions -------------------------------------------------
+
+expected<std::pair<uint16_t, const_byte_span>>
+framing::decode_closing_payload(const_byte_span payload) {
+  if (payload.size() == 0)
+    return std::make_pair(0, payload);
+  if (payload.size() < 2)
+    return make_error(caf::sec::protocol_error,
+                      "no status flag in closing payload");
+  auto status = (std::to_integer<uint_fast16_t>(payload[0]) << 8)
+                + std::to_integer<uint16_t>(payload[1]);
+  if (!detail::rfc3629::valid(payload.subspan(2)))
+    return make_error(sec::protocol_error,
+                      "malformed text message in closing payload");
+  if (status < 1000 || status >= 5000)
+    return make_error(sec::protocol_error,
+                      "invalid status code in closing payload");
+  // statuses between 3000 and 4999 are allowed and application specific
+  if (status >= 3000 && status < 5000)
+    return std::make_pair(status, payload.subspan(2));
+  // statuses between 1000 and 2999 are protocol defined
+  web_socket::status status_code;
+  auto success = from_integer(status, status_code);
+  if (!success)
```

Comment:
I can't find a reason to bind this result to a named variable, since it's only checked once:

```suggestion
  if (!from_integer(status, status_code))
```